Bottom: f85472204a6a85e02077990998033f19acba93fd
Top:    8b7c068f5e3f4959180c457cc431944c0d369b03
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 22:46:27 +0100

Separate out mark change notification from crosshair change notification


---

diff --git a/src/crosshair.c b/src/crosshair.c
index 9834577..06226d2 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -755,6 +755,26 @@ notify_crosshair_change (bool changes_complete)
     }
 }
 
+/* ---------------------------------------------------------------------------
+ * notify the GUI that data relating to the mark is being changed.
+ *
+ * The argument passed is false to notify "changes are about to happen",
+ * and true to notify "changes have finished".
+ *
+ * Each call with a 'false' parameter must be matched with a following one
+ * with a 'true' parameter. Unmatched 'true' calls are currently not permitted,
+ * but might be allowed in the future.
+ *
+ * GUIs should not complain if they receive extra calls with 'true' as parameter.
+ * They should initiate a redraw of the mark - which may (if necessary) mean
+ * repainting the whole screen if the GUI hasn't tracked the mark's location.
+ */
+void
+notify_mark_change (bool changes_complete)
+{
+  /* For now, just piggy back on the crosshair changes redrawing the mark */
+  notify_crosshair_change (changes_complete);
+}
 
 /* ---------------------------------------------------------------------------
  * switches crosshair on
@@ -806,6 +826,7 @@ HideCrosshair (void)
     }
 
   notify_crosshair_change (false);
+  notify_mark_change (false);
 }
 
 void
@@ -820,6 +841,7 @@ RestoreCrosshair (void)
     }
 
   notify_crosshair_change (true);
+  notify_mark_change (true);
 }
 
 /* ---------------------------------------------------------------------------
diff --git a/src/crosshair.h b/src/crosshair.h
index 671ae3d..efa6679 100644
--- a/src/crosshair.h
+++ b/src/crosshair.h
@@ -47,6 +47,7 @@
 #define	STATE_THIRD		2
 
 void notify_crosshair_change (bool changes_complete);
+void notify_mark_change (bool changes_complete);
 void CrosshairOn (void);
 void CrosshairOff (void);
 void HideCrosshair (void);
diff --git a/src/set.c b/src/set.c
index 16986e8..c77129e 100644
--- a/src/set.c
+++ b/src/set.c
@@ -342,20 +342,20 @@ SetLocalRef (LocationType X, LocationType Y, bool Showing)
 
   if (Showing)
     {
-      notify_crosshair_change (false);
+      notify_mark_change (false);
       if (count == 0)
 	old = Marked;
       Marked.X = X;
       Marked.Y = Y;
       Marked.status = true;
       count++;
-      notify_crosshair_change (true);
+      notify_mark_change (true);
     }
   else if (count > 0)
     {
-      notify_crosshair_change (false);
+      notify_mark_change (false);
       count = 0;
       Marked = old;
-      notify_crosshair_change (true);
+      notify_mark_change (true);
     }
 }
